### PR TITLE
MAT-4090 create FHIR resource for MADiE CqlLibrary

### PR DESCRIPTION
**Pull request by adongare**
## MADiE FHIR SERVICE

Jira Ticket: [MAT-4090](https://jira.cms.gov/browse/MAT-4090)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

**Review by gregory-akins — DISMISSED** [review]

**Review by RohitKandimalla — DISMISSED** [review]

File: src/main/java/gov/cms/madie/madiefhirservice/services/LibraryTranslatorService.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    attachments.add(createAttachment(CQL_CONTENT_TYPE, cql.getBytes()));
+    // TODO: XML & JSON contents
+    // attachments.add(createAttachment(XML_ELM_CONTENT_TYPE, elmXml.getBytes()));
+    // attachments.add(createAttachment(JSON_ELM_CONTENT_TYPE, elmJson.getBytes()));
+    return attachments;
+  }
+
+  private List<RelatedArtifact> distinctArtifacts(List<RelatedArtifact> artifacts) {
+    List<RelatedArtifact> result = new ArrayList<>(artifacts.size());
+    //Remove duplicates. I wish HapiFhir implemented equals. Today it is SadFhir for me.
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/madiefhirservice/services/LibraryTranslatorService.java]: Glad to see this comment will live on!

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -7,6 +7,8 @@ hapi:
   fhir:
     url: ${HAPI_FHIR_URL:http://localhost:8080/fhir}
 
+fhir-base-url: ${FHIR_BASEP_URL:http://ecqi.healthit.gov/ecqms}
```
Comment (jkotanchik-SB) [inline src/main/resources/application.yml]: What does the `P` in `BASEP` represent? Or is it a mt?

File: src/main/java/gov/cms/madie/madiefhirservice/services/LibraryTranslatorService.java (hunk 1 above)
Comment (adongare) [reply src/main/java/gov/cms/madie/madiefhirservice/services/LibraryTranslatorService.java]: I liked it 😄

File: src/main/resources/application.yml (hunk 2 above)
Comment (adongare) [reply src/main/resources/application.yml]: good catch, that is a typo.

**Review by jkotanchik-SB — APPROVED** [review]

**Pull request merged**